### PR TITLE
Apply black, pyupgrade and flake8 on stub files

### PR DESCRIPTION
**Pull request by s-weigand**
This PR changes the pre-commit config so that black, pyupgrade, and flake8 are also run on the `*.pyi` files.
It also adds a plugin for flake8 to improve the typing.

Since formatting rules for `*.pyi` files are different from those for normal `*.py` files some flake8 errors are ignored for those files.

In addition, a flake8 rule was added to ignore the unused imports error for `__init__.py` files, since those are only used to simplify the api.

### CODE REVIEW

File: glotaran/builtin/models/kinetic_spectrum/spectral_shape.pyi [hunk 1]
```diff
@@ -23,4 +23,4 @@ class SpectralShapeZero:
 
 class SpectralShape:
     @classmethod
-    def add_type(cls, type_name: str, type: Type) -> None: ...
+    def add_type(cls, type_name: str, type: type) -> None: ...
```
Comment (jsnel) [inline glotaran/builtin/models/kinetic_spectrum/spectral_shape.pyi:26]: ! type: type

File: glotaran/builtin/models/kinetic_image/irf.pyi [hunk 2]
```diff
@@ -1,50 +1,31 @@
...
-    def width(self) -> Parameter:
-        ...
-
+    def width(self) -> Parameter: ...
 
 class Irf:
     @classmethod
-    def add_type(cls, type_name: str, type: Type) -> None:
-        ...
+    def add_type(cls, type_name: str, type: type) -> None: ...
```
Comment (jsnel) [inline glotaran/builtin/models/kinetic_image/irf.pyi:31]: ! type: type

Comment (s-weigand) [reply glotaran/builtin/models/kinetic_image/irf.pyi]: Didn't look too closely at the changes since they were autogenerated. Renamed it!

**Review by jsnel — APPROVED** [review]
LGTM

**Pull request merged**